test/runner.js

```diff
@@ -80,17 +94,42 @@ files = files.reduce( ( memo, filePath ) => {
 }, [] );
 
 if ( program.nodeTotal > 1 ) {
-	files = files.filter( ( file, index ) => {
+	testFiles = testFiles.filter( ( file, index ) => {
 		return index % program.nodeTotal === program.nodeIndex;
 	} );
 }
 
-files.forEach( setup.addFile );
-
-mocha.addFile( path.join( __dirname, 'load-suite.js' ) );
+let runner = null;
+const runMocha = () => {
+	console.log( chalk.green( 'Running tests at: ' ),
+		chalk.yellow( new Date() ) );
 
-mocha.run( function( failures ) {
-	process.on( 'exit', function() {
-		process.exit( failures ); //eslint-disable-line no-process-exit
+	runner = getMocha().run( () => {
+		runner = null;
 	} );
-} );
+};
+
+if ( program.watch ) {
+	console.log( chalk.green( 'Watch mode enabled' ) );
+
+	const watchFiles = glob.sync( '**/*.@(js|jsx)', { ignore: 'node_modules/**' } );
```

Comment:
I see now that it won't cover the scenario when I create new test file. Do they even care about it in Mocha default test runner? I'm not saying we must have it, just curious :D